docs/getting-started/0-agent.md

```diff
@@ -28,6 +31,185 @@ const agentConfig: InitConfig = {
+  poolName: 'BCovrin Test',
+  genesisTransactions: BCOVRIN_TEST_GENESIS,
+  logger: new ConsoleLogger(LogLevel.debug),
+}
+
+const agent = new Agent(agentConfig, agentDependencies)
+```
+
+### 5- Create Transports
+
+After creating the Agent object, we need to create and set the Inbound/Outbound transports that will handle traffic to/from the Agent.
+
+```ts
+const httpOutboundTransporter = new HttpOutboundTransporter()
+agent.registerOutboundTransport(httpOutboundTransporter)
+
+//Inbound transports are currently built-in, you don't need to create them.
+```
+
+More about [Transports](1-transports.md).
+
+### 6- Init Agent
+
+After creating the Agent object and configuring it, we initialize the Agent.
+
+```ts
+// It's highly recommended to wrap the initialization flow in a try/catch block
+try {
+  await agent.initialize()
+  console.log('Initialized agent!')
+} catch (e) {
+  console.log(`Agent init error:${e}`)
+}
+```
+
+### 7- Handling State Changes
+
+After you successfully initialize your Agent, you will notice that most of the hard work is being done by the underlying Aries/Indy framework. However, you as a controller will need to intercept in some situations when there is a state change (like when you need to accept an offered credential or reject/accept an incoming connection request). This is done through state change handlers.
+
+#### Creating Event(State) Handlers
+
+```ts
+agent.events.on<AgentMessageReceivedEvent>(AgentEventTypes.AgentMessageReceived, handleBasicMessageReceive)
+
+agent.events.on<CredentialStateChangedEvent>(
+  CredentialEventTypes.CredentialStateChanged,
+  // Custom callback for handling any state change when offering/receiving VCs
+  handleCredentialStateChange
+)
+
+agent.events.on<ConnectionStateChangedEvent>(
+  ConnectionEventTypes.ConnectionStateChanged,
+  // Custom callback for handling any state change when offering/receiving a connection
+  handleConnectionStateChange(event)
+)
+```
+
+Example: This sample credential callback shows how to detect that a credential is received, show the user the credential asserts and give the option to accept/reject the offered credential.
+
+```ts
+const handleCredentialStateChange = async (event: CredentialStateChangedEvent) => {
+  console.log(
+    `>> Credential state changed: ${event.payload.credentialRecord.id}, previous state -> ${event.payload.previousState} new state: ${event.payload.credentialRecord.state}`
+  )
+
+  if (event.payload.credentialRecord.state === CredentialState.OfferReceived) {
+    console.log(`>> Received offer, should display credential to user`)
+
+    // Display offer to user
+    // On user click "Accept"
+    console.log(`>>ACCEPTING OFFER`)
+    agent.credentials.acceptOffer(event.payload.credentialRecord.id)
+  } else if (event.payload.credentialRecord.state === CredentialState.Done) {
+    Alert.alert('Credentail Saved')
+  }
+}
+```
+
+More about [Credentials](5-credentials.md).
```

Comment:
Could you reformat these two sentences? Does not look nice when formatted.

(Maybe add a section for related subjects at the end?

Reply:
this is related to the block itself and shouldn't be added in a separate title I think